docs/pluggable-packages.md

```diff
@@ -0,0 +1,124 @@
+
+It's also possible to pull the required contents from referenced remote repositories outside the cluster, then serve them in the Git server. This approach requires more work because we now need to render manifests instead of offloading it to ArgoCD.
+
+
+## Future improvements
+
+- Extend support for using different tools for CD and imperative workflows. For example, we may consider supporting Tekton or Flagger for running imperative workflows.
+- Extend support for other GitOps solutions such as Flux CD.
+- Referencing remote private repositories.
+- Add support for authentication. As use cases grow, it will be necessary to support authentication mechanisms. Example use cases: GitHub credentials for ArgoCD, AWS credentials to pull images from a ECR registry.
+
+## Alternatives Considered
+
+#### Use OCI images as applications
+
+Projects such as Sealer and Kapp aim to use OCI images as the artifact to define and deploy multiple Kubernetes resources. This has a few advantages. 
+
+- Immutable single artifact that can be used to deploy to different clusters.
+- Simplicity. Application dependencies and supporting resources are defined and confined in the image.
+- Can use standard Kubernetes YAML files, Helm Charts, and Kustomize. 
+- Native support for signing and verification through industry standard tools like cosign.
+
+In addition, both tools support applying changes in particular order. For example, you can run a Kubernetes Job to migrate database schema before rolling out a new image. 
+
+Since many of the goals are covered by both projects, it is possible to incorporate some of their tools and libraries to implement our goals. 
+
+While this approach addresses most of our goals, there are drawbacks.
+
+Firstly, it introduces a new layer for end users to debug. For end users to debug an issue related to Kubernetes manifest rendering, they now need to:
+1. Figure out which OCI image contains the package with the problem. The problem may reside in one of dependent OCI images. 
+2. Extract the contents of the OCI image. Determine which one file is responsible for the issue. 
+3. Correct the issue.
+4. Publish a new image.
+
+Secondly, these tools and using OCI images as packages are not well adopted by CNOE members as evidenced by the tech radar. idpbuilder should be useful and relevant to CNOE members.
+
+#### Define specs for imperative work
+
+To support use cases where more complex steps are needed to orchestrate different services in to the cluster, idpbuilder could define a configuration spec where end users can define their own steps. For example, a spec may look something like the following. 
+
+```yaml
+apiVersion: idpbuilder.cnoe.io/v1alpha1
+kind: Config
+metadata:
+  name: test
+spec:
+  packages:
+    - name: crossplane
+      preCreation:
```

Comment:
We need as additional fields:
- version
- description
- deployment_type: helm, kustomize, etc
- url: => external repository hosting the chart, oci image, raw resources, etc
- values: object including the values to be passed to the helm chart of yaml config file passed as parameter and mounted as volume

Reply:
I'll have to put more thoughts into this later, but my point is we should avoid creating another config format if possible.